calculator.js

```diff
@@ -1,146 +1,192 @@
-    calculator.current = number.slice(0,-1);
-    livesResult.innerHTML = calculator.current;
-
-    if(calculator.current.length === 0) {
+  
+  if (character === DELETE && calculator.term.length > 0) {
+    const firstTerm       = calculator.term;
+    const number          = calculator.current;
+    calculator.term       = firstTerm.slice(0,-1);
+    calculator.current    = number.slice(0,-1);
+    display.innerHTML     = calculator.term;
+    
+    if (calculator.current.length === 0) {
+      display.innerHTML     = 0;
       livesResult.innerHTML = 0;
     }
   }
 };
-numbers.forEach(function(input) {
+
+/*
+ *PRE-CONDITIONS: It needed two terms.
+ *POST-CONDITIONS:This function return the final result.
+ */
+function solution(first, second, operator) {
+
+  const a = parseInt(first, BASE);
+  const b = parseInt(second, BASE);
+  var total;
+
+  switch (operator) {
+    case PLUS:
+      total = a + b;
+      break;
+    case SUBTRACT:
+      total = a - b;
+      break;
+    case MULTIPLY:
+      total = a * b;
+      break;
+    case DIVIDE:
+      if (b === 0) {
+        total = "FATAL ERROR";
+      } else {
+        total = a / b;
+      }
+      break;
+  }
+  return total || first;
+}
+
+/*PRE-CONDITIONS:
+ *POST-CONDITIONS: It Takes the elements from the global varables and solves the operations
+ in order.
+ */
+const resolution = function() {
+  var first, second, operator, total;
+  var i = 0;
+  calculator.container.push(calculator.current);
+  console.log(calculator.container);
+
+  for (i ; i < calculator.container.length; i++) {
+    if (i === 0) {
+      first    = calculator.container[i];
+      i++;
+      operator = calculator.container[i];
+      i++;
+      second   = calculator.container[i];
+      total    = solution(first,second, operator);
+    } else {
+      first    = total;
+      operator = calculator.container[i];
+      i++;
+      second   = calculator.container[i];
+      total    = solution(first,second, operator);
+    }
+    console.log(first);
+    console.log(operator);
+    console.log(second);
+  }
+  console.log(total);
+  console.log("------------");
+  calculator.term            += calculator.current;
+  livesResult.innerHTML       = total;
+  display.innerHTML = ' ';
+  calculator.term             = total;
+  calculator.container.length = 0;
+  calculator.current          = total;
+}
+
+numbers.forEach(function (input) {
   input.addEventListener("click", saveNumber)
 });
-operators.forEach(function(operators) {
+
+operators.forEach(function (operators) {
   operators.addEventListener("click", operate)
 });
-actions.forEach(function(actions) {
-  actions.addEventListener("click", doActions)
+
+actions.forEach(function (actions) {
+  actions.addEventListener("click", wipeAction)
+});
+
+equal.addEventListener('click', resolution);
+
+document.addEventListener('keydown', function (event) {
+  let keyName = event.key;
+
+  if (!isNaN(keyName)) {
+    saveNumber(EMPTY, keyName);
```

Comment:
Probaste que pasa si en vez de enviar EMPTY envias el event?
```suggestion
    saveNumber(event, keyName);
```